### PR TITLE
Queries and procedures

### PR DESCRIPTION
**Pull request by Leen-odeh3**

### CODE REVIEW

File: Scripts/ListofBorrowedBooks.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-- 1. List of Borrowed Books
+-- Retrieve all books borrowed by a specific borrower, including those currently unreturned.
+declare @BorrowerId int = 5;
+
+SELECT  Library.Books.Title, Books.Author, Loans.DateBorrowed, Loans.DueDate
+FROM Library.Loans
```
Comment (ahmad-m-abbas) [inline Scripts/ListofBorrowedBooks.sql]: I suggest aliasing here, more readable

File: Scripts/BorrowingFrequency.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- 3. Borrowing Frequency using Window Functions
+--  Rank borrowers based on borrowing frequency.
+WITH BorrowerFrequency AS (
+    SELECT
+        BorrowerID,
+        COUNT(*) AS BorrowingFrequency,
+        ROW_NUMBER() OVER (ORDER BY COUNT(*) DESC) AS BorrowerRank
```
Comment (ahmad-m-abbas) [inline Scripts/BorrowingFrequency.sql]: Just a note, `ROW_NUMBER()`  assigns a unique rank to each borrower even if two borrowers have the same borrowing frequency it we want to make the same frequency to have the same rank and skip subsequent ranks, consider using `RANK()` or `DENSE_RANK()` instead.

File: Scripts/Add New Borrowers.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
+-- 5. Stored Procedure
+-- Procedure Name: sp_AddNewBorrower
+CREATE OR ALTER PROCEDURE sp_AddNewBorrower
```
Comment (ahmad-m-abbas) [inline Scripts/Add New Borrowers.sql:6]: NIT: May delete it at first then create

File: Scripts/CalculateOverdueFees.sql [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    @LoanID INT
+)
+RETURNS DECIMAL(10, 2)
+AS BEGIN
+    DECLARE @DueDate DATE;
+    DECLARE @DateReturned DATE;
+    DECLARE @OverdueDays INT;
+    DECLARE @OverdueFee DECIMAL(10, 2);
+
+    SELECT @DueDate = DueDate, @DateReturned = DateReturned
```
Comment (ahmad-m-abbas) [inline Scripts/CalculateOverdueFees.sql:16]: check the `@DueDate` if null

File: Scripts/OverdueAnalysis.sql [hunk 5]
```diff
@@ -0,0 +1,31 @@
+-- 8. Overdue Analysis
+--  List all books overdue by more than 30 days with their associated borrowers
+DECLARE @OverdueDays INT = 10;
+WITH OverdueBooks AS (
+    SELECT
+        BookID,
+        BorrowerID,
+	   	DateBorrowed,
```
Comment (ahmad-m-abbas) [inline Scripts/OverdueAnalysis.sql]: fix format

File: Scripts/OverdueAnalysis.sql [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+)
+SELECT
+    k.BookID,
+    k.Title,
+    k.Author,
+    w.BorrowerID,
+    w.FirstName,
+    w.LastName,
+    w.Email,
+	o.DateBorrowed,
```
Comment (ahmad-m-abbas) [inline Scripts/OverdueAnalysis.sql]: why the same issue lol

**Review by ahmad-m-abbas — APPROVED** [review]

File: Scripts/Add New Borrowers.sql (hunk 3 above)
Comment (Leen-odeh3) [reply Scripts/Add New Borrowers.sql:6]: I added " ALTER " if  "sp_AddNewBorrower" EXISTS modify It is similar to create , true?

**Pull request merged**